File: frontend/src/Models/BackendDtos/Kanban/BoardGetDto.ts
import type { CardListGetDto } from "./CardListGetDto.ts";
import type {LabelGetDto} from "./LabelGetDto.ts";

export type BoardGetDto = {
    boardId: number;
    position?: string;
    boardName: string;
    ownedByUserId: string;
    cardLists: CardListGetDto[];
    labels: LabelGetDto[];
}

File: backend/ArhiTodo.Domain/Entities/DTOs/BoardGetDto.cs
namespace ArhiTodo.Domain.Entities.DTOs;

public class BoardGetDto
{
    public int BoardId { get; set; }
    
    public required string Position { get; set; }
        
    public required string BoardName { get; set; }

    public required Guid OwnedByUserId { get; set; }
    
    public List<CardListGetDto> CardLists { get; set; } = [];
    public List<LabelGetDto> Labels { get; set; } = [];
}